src/nvim/api/ui_events.in.h

```diff
@@ -68,4 +68,15 @@ void popupmenu_select(Integer selected)
 void tabline_update(Tabpage current, Array tabs)
     FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
 
+void cmdline_enter(void)
+    FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
+void cmdline_show(Array content, Integer pos, String firstc, String prompt, Integer level)
+    FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
+void cmdline_pos(Integer pos, Integer level)
+    FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
+void cmdline_char(String c, Integer shift)
```

Comment:
can cmdline_pos and cmdline_char be combined into a general cmdline_update event (which may send other updates as well, in a map-like form)